src/exchangeRateRouter.ts

```diff
@@ -25,7 +32,21 @@ const asExchangeRatesReq = asObject({
   data: asArray(asExchangeRateReq)
 })
 
+const asExtendedReq = <T>(asExtension: CleanerShape<T>) => (
+  raw: any
+): express.Request & T => {
+  const extension = asObject(asExtension)(raw)
+  Object.assign(raw, extension)
+  return raw
+}
```

Comment:
Copy and paste this to a `util/asExtendedReq.ts` file. This way it's a reusable utility throughout this project.